### PR TITLE
Move protocol and reset if contradictory

### LINKED ISSUE #625: Warn users when https is enabled and port is 80
reef-pi allow users to configure api/UI address (host and port) as well http/https. Currently this allow users to specify port 80 (http) while https is enabled. When configured like this user will see the webpage being download instead of rendering it. The following screenshot shows port 443 is used without https being enabled. 
screen shot 2018-10-28 at 8 13 02 am

As a mitigation measure, lets implement warning message in configuration UI (validation?) when user specifies port thats not 80 for http , and port 443 for https.

### PR DESCRIPTION
**Pull request by Ranthalion**
## Description

Moves https to the address field and removes contradictory settings on change.  Fix is as described in issue comments.

## Related Issue

Fixes #625

### CODE REVIEW

File: front-end/src/configuration/settings.jsx [hunk 1]
```diff
@@ -70,6 +73,45 @@ class settings extends React.Component {
...
+
+  handleSetProtocolHttp () {
+    this.handleSetProtocol(false)
+  }
+
+  handleSetProtocolHttps () {
+    this.handleSetProtocol(true)
+  }
+
+  handleSetProtocol (value) {
```
Comment (zekth) [inline front-end/src/configuration/settings.jsx]: Is it possible to add a test for this?

Comment (Ranthalion) [reply front-end/src/configuration/settings.jsx]: Good point. I'll work on adding tests for this tab tonight.

Comment (Ranthalion) [reply front-end/src/configuration/settings.jsx]: A few tests have been added for the new functionality.

**Comment by ranjib** [comment]
Thank you :-)

**Pull request merged**